controllers/bsl.go

```diff
@@ -377,14 +384,53 @@ func (r *DPAReconciler) validateProviderPluginAndSecret(bslSpec velerov1.BackupS
 	return nil
 }
 
-func (r *DPAReconciler) ensureBackupLocationHasVeleroOrCloudStorage(dpa *oadpv1alpha1.DataProtectionApplication) error {
-	for _, bsl := range dpa.Spec.BackupLocations {
-		if bsl.CloudStorage == nil && bsl.Velero == nil {
-			return fmt.Errorf("no bucket CloudStorage or velero BackupStorageLocation provided for backupLocations")
-		}
+func (r *DPAReconciler) ensureBackupLocationHasVeleroOrCloudStorage(bsl *oadpv1alpha1.BackupLocation) error {
+	if bsl.CloudStorage == nil && bsl.Velero == nil {
+		return fmt.Errorf("no bucket CloudStorage or velero BackupStorageLocation provided for backupLocations")
```

Comment:
For consistency, I would change message here to **"BackupLocation must have velero or bucket configuration"**

Since this check was in validator.go, it would run prior to this. So, I think this message was never presented to the user

Reply:
fixed.